lib/awful/keyboard.lua

```diff
@@ -0,0 +1,77 @@
+---------------------------------------------------------------------------
+--- Utilities related to the keyboard and text entry.
+--
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2018 Emmanuel Lepage Vallee
+-- @module awful.keyboard
+---------------------------------------------------------------------------
+local capi = {root = root, client = client}
+
+local module = {}
+
+-- The hash order isn't consistent when the content changes, but it is
+-- constant when the table doesn't change.
+local conversion = {
+    Super_L   = "Mod4",
+    Control_L = "Control",
+    Shift_L   = "Shift",
+    Alt_L     = "Mod1",
+    Super_R   = "Mod4",
+    Control_R = "Control",
+    Shift_R   = "Shift",
+    Alt_R     = "Mod1",
+}
```

Comment:
Sorry, but no. This is (or at least: can be) keyboard layout specific. This is not true in general. Please don't make me hunt a keyboard layout where this actually breaks, but if you really want this kind of information, you'll have to figure out how XKB specifies keyboard layouts and how to query this information from the layout. I bet everything you'd need is in libxkbcommon, but I do not really know the details.